tools/src/vendoring/legacy.ts

```diff
@@ -25,6 +25,7 @@ interface VendoredModuleUpdateStep {
   /**
    * Hook that is fired by the end of vendoring an Android file.
    * You should use it to perform some extra operations that are not covered by the main flow.
+   * @deprecated Use {@link VendoredModuleConfig.moduleModifier} instead.
```

Comment:
do you mean we should deprecate `onDidVendorAndroidFile` but to write custom logic in moduleModifier before copying all the files?

Reply:
Yeah, it would be better to have all such logic happening in a single place 😉 
AFAIK, I was the person that suggested `onDidVendorAndroidFile` and at that time I was not aware of `VendoredModuleConfig.moduleModifier` 😅 
It's all legacy vendoring anyway, so I guess it's a matter of time we migrate to the new one.

Reply:
it's me to introduce the `onDidVendorAndroidFile`, because `VendoredModuleConfig.moduleModifier` is easier to do the [file rename](https://github.com/expo/expo/blob/a03d018db9dbc03c6ae9c866a613e1c182413aba/tools/src/vendoring/legacy.ts#L294-L318).
anyway, to vendor @shopify/react-native-skia, i am having a basic new vendoring script for android. hopefully we can remove the legacy vendoring in the future.